ingestion_tools/scripts/common/image.py

```diff
@@ -446,6 +479,53 @@ def has_label(self) -> bool:
             return bool(np.any(self.volume_reader.get_pyramid_base_data() == self.label))
 
 
+class MultiLabelMaskConverter(TomoConverter):
+    def __init__(
+        self,
+        fs: FileSystemApi,
+        filename: str,
+        header_only: bool = False,
+        scale_0_dims: tuple[int, int, int] | None = None,
+    ):
+        super().__init__(fs=fs, filename=filename, header_only=header_only, scale_0_dims=scale_0_dims)
+
+    def get_pyramid_base_data(self) -> np.ndarray:
+        data = self.volume_reader.get_pyramid_base_data()
+
+        if not self.scale_0_dims:
+            return self.scaled_data_transformation(data)
+
+        from scipy.ndimage import zoom
+
+        x, y, z = data.shape
+        nx, ny, nz = self.scale_0_dims
+        zoom_factor = (nx / x, ny / y, nz / z)
+        if zoom_factor == (1.0, 1.0, 1.0):
+            return self.scaled_data_transformation(data)
+
+        # rescaled = rescale(
+        #     data,
+        #     scale=zoom_factor,
+        #     order=0,
+        #     preserve_range=True,
+        #     anti_aliasing=False,
+        # )
+
+        rescaled = zoom(data, zoom=zoom_factor, order=0)
+
+        return self.scaled_data_transformation(rescaled)
+
+    @classmethod
+    def scaled_data_transformation(cls, data: np.ndarray) -> np.ndarray:
+        # For instance segmentation masks we have multiple labels, so we want an uint 16 output.
+        # We used uint16 and not uint32 as it seems MRC format doesn't handle well int > 16.
+        # downscale_local_mean will return float array even for bool input with non-binary values
```

Comment:
Is the downscale_local_mean comment here still relevant? using block_reduce now I think